### DISCUSSION TITLE
Event Trait

### DISCUSSION THREAD

**Discussion by amirHdev** [DioxusLabs/dioxus#3124] [Q&A]
Is there event trait to use it in handlers and callers ? 

For example we have modal and we want to use keyboard event and mouse event as on click event to close it . By this we should define 2 function with same process.

**Comment by ryo33** [comment]
There is an `Event` struct, and it has a generic field `T`, so it seems we could define a function like that:

```rust
fn handle_event<T>(event: Event<T>) {
    println!("event happened!);
    event.stop_propagation();
}
```

And it should be usable as something like:

```rust
component {
    onclick: move |evt: Event<MouseData>| {
        handle_event(evt);
    },
    onkeydown: move |evt: Event<KeyboardData>| {
        handle_event(evt);
    },
}
```

**Comment by amirHdev** [comment]
@ryo33 generic is choice but not while we want to pass signals through the function or borrow . we need closure function and we have to create function to through out implementation of closure with generic argument and obviously lifetime .

event trait is better choice if exists in this repo .
